### PR TITLE
Fixes bug where save buffer is not updated when entering view only

### PR DESCRIPTION
**Pull request by jSwanson99**

### CODE REVIEW

File: src/components/SceneConfigMenu.js [hunk 1]
```diff
@@ -222,14 +222,18 @@ class ConfigModal extends Component {
 
   // Toggles whether the editor is showing
   viewToggle = () => {
-    let style = this.props.scene.settings.viewOnly ? btnStyle.off : btnStyle.on;
+    let viewOnly = this.props.scene.settings.viewOnly
```
Comment (jasondkiesling) [inline src/components/SceneConfigMenu.js]: This is not technically wrong, but it adds unnecessary complexity to the code. It is best to only use a variable to store a prop when you're using it multiple times or in long strings. In addition, props stored as a variable should always be a const as you can't change them.

File: src/components/View.js [hunk 2]
```diff
@@ -171,33 +170,35 @@ class View extends Component {
 
   render = () => {
     return (
-      <a-scene physics="debug: false; friction: 3; restitution: .3;" embedded debug="false">
```
Comment (jasondkiesling) [inline src/components/View.js]: What's up with these changes?

File: src/components/layouts/Ide.js [hunk 3]
```diff
@@ -6,46 +6,60 @@ import View from '../View';
...
+                    projects={this.props.projects}
+                    courses={this.props.courses}
+                    classroomActions={this.props.classroomActions}
+                    classrooms={this.props.classrooms}
+                    layoutType={layoutTypes.IDE}
+                />
+
+                <div className="row no-gutters">
+                    {
+                        /*
```
Comment (jasondkiesling) [inline src/components/layouts/Ide.js]: Let's move comments about issues to Github. Code comments are best when they more about what the code does or quick issues. Feel free to make a reference to Github by citing the issue number.

File: src/components/layouts/Ide.js [hunk 4]
```diff
@@ -6,46 +6,60 @@ import View from '../View';
 
 import * as layoutTypes from '../../constants/LayoutTypes.js';
 
-export const Ide = ({ editor, editorActions, user, authActions, scene, sceneActions, projectActions, courseActions, projects, courses, match, classroomActions, classrooms }) => (
```
Comment (jasondkiesling) [inline src/components/layouts/Ide.js]: There are also a lot of format changes here. What are these for?

File: src/components/Editor.js [hunk 5]
```diff
@@ -10,6 +10,12 @@ import customCompleter from './customCompleter.js'
 *
 */
 class Editor extends Component {
+  componentWillUnmount() {
+    //Text is saved in state when editor unmounts / when in view only.
+    //This does not fully 'save' it because state will be lost when we
```
Comment (jasondkiesling) [inline src/components/Editor.js]: Although this is a valid, this is something that's I'm working on fixing. This comment can be simplified to say that it updates the reducer state before closing the editor.

File: src/components/View.js (hunk 2 above)
Comment (jSwanson99) [reply src/components/View.js]: These are just whitespace changes I didn't intentionally make, is there a way to ignore/revert these easily?

**Pull request closed without merging**